### DISCUSSION TITLE
How to improve Foxglove's screen refresh rate

### DISCUSSION THREAD

**Discussion by zixuan-su** [foxglove/community#31] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/64144933/183797010-180ee246-1ce0-4e2a-975c-09af88af18bc.png)

My Foxglove screen refresh rate is only about 17 Hz , what should I do to improve it?

**Comment by jtbandes** [comment, marked as answer]
This rate is not the screen refresh rate, but is calculated from the timestamps of the mcap file, or the rate of data received from the publisher.

**Reply by zixuan-su** [reply]
> This rate is not the screen refresh rate, but is calculated from the timestamps of the mcap file, or the rate of data received from the publisher.
Thanks for your reply, and I would like to ask if I can detect the software frame rate

**Reply by jtbandes** [reply]
Try this panel: https://foxglove.dev/docs/studio/panels/studio-playback-performance

The 3D Beta panel also has a render performance indicator.

Both of these indicators are only an estimate and you shouldn't assume they are 100% accurate.

<img width="358" alt="image" src="https://user-images.githubusercontent.com/14237/183809774-6e3e92d2-9323-4d89-8930-50e252d56dab.png">

**Reply by zixuan-su** [reply]
> Try this panel: https://foxglove.dev/docs/studio/panels/studio-playback-performance
> 
> The 3D Beta panel also has a render performance indicator.
> 
> Both of these indicators are only an estimate and you shouldn't assume they are 100% accurate.
> 
> <img alt="image" width="358" src="https://user-images.githubusercontent.com/14237/183809774-6e3e92d2-9323-4d89-8930-50e252d56dab.png">

Thanks， I'll look into how to do it
